### PR TITLE
Use touch state from ft63x6 driver.

### PR DESCRIPTION
**Pull request by nielsnl68**
# What does this implement/fix?

This is an reworked version for the FT63x6 driver. The old method did not registered correctly when the touch was released.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [x] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040
- [ ] BK72xx
- [ ] RTL87xx

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/ft63x6/ft63x6.cpp [hunk 1]
```diff
@@ -40,26 +40,11 @@ void FT63X6Touchscreen::setup() {
...
-  this->x_raw_max_ = 320;
-  this->y_raw_max_ = 480;
-}
-
-void FT63X6Touchscreen::update_touches() {
-  int touch_count = this->read_touch_count_();
-  if (touch_count == 0) {
-    return;
+  if (this->x_raw_max_ == this->x_raw_min_) {
+    this->x_raw_max_ = 319;
```
Comment (gpambrozio) [inline esphome/components/ft63x6/ft63x6.cpp]: From what I can tell, max value is exclusive, so I think this should still be 320.

File: esphome/components/ft63x6/ft63x6.cpp [hunk 2]
```diff
@@ -40,26 +40,11 @@ void FT63X6Touchscreen::setup() {
...
-  int16_t y = this->read_touch_coordinate_(FT63X6_ADDR_TOUCH1_Y);
-  this->add_raw_touch_position_(touch_id, x, y);
-
-  if (touch_count >= 2) {
-    touch_id = this->read_touch_id_(FT63X6_ADDR_TOUCH2_ID);  // id2 = 0 or 1(~id1 & 0x01)
-    x = this->read_touch_coordinate_(FT63X6_ADDR_TOUCH2_X);
-    y = this->read_touch_coordinate_(FT63X6_ADDR_TOUCH2_Y);
-    this->add_raw_touch_position_(touch_id, x, y);
+  if (this->y_raw_max_ == this->y_raw_min_) {
+    this->y_raw_max_ = 459;
```
Comment (gpambrozio) [inline esphome/components/ft63x6/ft63x6.cpp]: I believe this should be 480

**Comment by nielsnl68** [comment]
@gpambrozio you are right. I doubted myself earlier :(

**Review by gpambrozio — APPROVED** [review]

File: esphome/components/ft63x6/ft63x6.cpp [hunk 3]
```diff
@@ -76,23 +61,33 @@ void FT63X6Touchscreen::dump_config() {
...
-// Touch functions
-uint16_t FT63X6Touchscreen::read_touch_coordinate_(uint8_t coordinate) {
-  uint8_t read_buf[2];
-  read_buf[0] = this->read_byte_(coordinate);
-  read_buf[1] = this->read_byte_(coordinate + 1);
-  return ((read_buf[0] & 0x0f) << 8) | read_buf[1];
-}
-uint8_t FT63X6Touchscreen::read_touch_id_(uint8_t id_address) { return this->read_byte_(id_address) >> 4; }
+  if (!this->read_bytes(0x00, (uint8_t *) data, 15)) {
+    esph_log_e(TAG, "Failed to read touch data");
```
Comment (jesserockz) [inline esphome/components/ft63x6/ft63x6.cpp]:
```suggestion
    ESP_LOGE(TAG, "Failed to read touch data");
```

File: esphome/components/ft63x6/ft63x6.cpp [hunk 4]
```diff
@@ -76,23 +61,33 @@ void FT63X6Touchscreen::dump_config() {
...
 
-uint8_t FT63X6Touchscreen::read_byte_(uint8_t addr) {
-  uint8_t byte = 0;
-  this->read_byte(addr, &byte);
-  return byte;
+  if (((data[FT63X6_ADDR_TOUCH1_STATE] >> 6) & 0x01) == 0) {
+    touch_id = data[FT63X6_ADDR_TOUCH1_ID] >> 4;  // id1 = 0 or 1
+    x = encode_uint16(data[FT63X6_ADDR_TOUCH1_X] & 0x0F, data[FT63X6_ADDR_TOUCH1_X + 1]);
+    y = encode_uint16(data[FT63X6_ADDR_TOUCH1_Y] & 0x0F, data[FT63X6_ADDR_TOUCH1_Y + 1]);
+    // ESP_LOGV(TAG, "Add Touch A: id:%3d (x:%4d, y:%4d) %x", touch_id, x, y, (data[0x03] >> 6) & 0x01);
```
Comment (jesserockz) [inline esphome/components/ft63x6/ft63x6.cpp]: No commented code please
```suggestion
```

File: esphome/components/ft63x6/ft63x6.cpp [hunk 5]
```diff
@@ -76,23 +61,33 @@ void FT63X6Touchscreen::dump_config() {
...
+    this->add_raw_touch_position_(touch_id, x, y);
+  }
+  if (((data[FT63X6_ADDR_TOUCH2_STATE] >> 6) & 0x01) == 0) {
+    touch_id = data[FT63X6_ADDR_TOUCH2_ID] >> 4;  // id1 = 0 or 1
+    x = encode_uint16(data[FT63X6_ADDR_TOUCH2_X] & 0x0F, data[FT63X6_ADDR_TOUCH2_X + 1]);
+    y = encode_uint16(data[FT63X6_ADDR_TOUCH2_Y] & 0x0F, data[FT63X6_ADDR_TOUCH2_Y + 1]);
+    // ESP_LOGV(TAG, "Add Touch B: id:%3d (x:%4d, y:%4d) %x", touch_id, x, y, (data[0x09] >> 6) & 0x01);
```
Comment (jesserockz) [inline esphome/components/ft63x6/ft63x6.cpp]:
```suggestion
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

**Review by jesserockz — APPROVED** [review]

**Pull request merged**